docs/src/arrays.md

```diff
@@ -0,0 +1,219 @@
+# Arrays
+
+When the inputs and outputs of the functions are arrays (potentially with scalars), a modified version of Giles' method is a quick way of deriving `frule`s and `rrule`s.
+Giles' method is succinctly explained in [this paper](https://people.maths.ox.ac.uk/gilesm/files/NA-08-01.pdf), but we will slightly generalize it.
+
+## Forward mode rules
+
+Given a function
+
+```julia
+C = f(A::Array, B::Array)::Array
+```
+
+we write the differential of the output in terms of differentials of the inputs as
+
+$$dC = \frac{\partial f}{\partial A} dA + \frac{\partial f}{\partial B} dB.$$
+
+The pushforward is obtained by dividing by the differential wrt a single scalar input $t$.
+Writing $\dot{C} = \frac{dC}{dt}$, we replace the differentials with the forward mode sensitivities:
+
+$$\dot{C} = \frac{\partial f}{\partial A} \dot{A} + \frac{\partial f}{\partial B} \dot{B}.$$
+
+The terms $\frac{\partial f}{\partial A}$ are array-array derivatives, i.e. a type of Jacobian.
+We do not write these down explicitly, but we instead use differential identities to write down the terms $\frac{\partial f}{\partial A} dA$, which as we've seen behave like the Jacobian-vector-products $\frac{\partial f}{\partial A} \dot{A}$.
+The differential identities follow directly from the usual scalar identities.
+
+### Matrix addition
+
+$$C = A + B$$
+
+$$dC = dA + dB$$
+
+### Matrix multiplication
+
+$$C = A B$$
+
+First we write in component form.
+
+$$C_{ij} = \sum_k A_{ik} B_{kj}$$
+
+Then we apply the product rule to get the scalar differential identity.
+
+$$dC_{ij} = \sum_k dA_{ik}\ B_{kj} + A_{ik}\ dB_{kj}$$
+
+But this is just a sum matrix products:
+
+$$C = dA\ B + A\ dB$$
+
+So we now have the matrix product rule.
+
+### Matrix inversion
+
+$$C A = A C = I$$
```

Comment:
For the sake of flow, maybe start by stating `C := inv(A)` or whatever to make it clear what function the rule is being derived for. When you're reading these derivations for the first time I think it's helpful to make it clear what the input is and what the output is so that you don't get lost.

Reply:
What do you think about just using `A^{-1}` instead of `C`?